### PR TITLE
Add support for an optional local consul agent

### PR DESCRIPTION
**Pull request by rvandegrift**
In some environments, it's useful to have a consul agent for vault.
This patch adds an optional agent to vault's pod.  It is can be turned
on by defining consulAgent.join to point to a consul server.

### CODE REVIEW

File: incubator/vault/values.yaml [hunk 1]
```diff
@@ -6,6 +6,21 @@ image:
   repository: vault
   tag: 0.9.0
   pullPolicy: IfNotPresent
+consulAgent:
+  repository: consul
+  tag: 1.0.6
```
Comment (mattfarina) [inline incubator/vault/values.yaml]: 1.0.7 is now out. Can that newer version be used instead?

File: incubator/vault/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: A Helm chart for Vault, a tool for managing secrets
 name: vault
-version: 0.4.3
+version: 0.4.4
```
Comment (mattfarina) [inline incubator/vault/Chart.yaml]: Because there are feature additions can you bump this to `0.5.0` for the version

**Review by mattfarina — COMMENTED** [review]
/cc @jpds

File: incubator/vault/values.yaml (hunk 1 above)
Comment (rvandegrift) [reply incubator/vault/values.yaml]: Yep - I've tested 1.0.7 and updated the default tag

File: incubator/vault/Chart.yaml (hunk 2 above)
Comment (rvandegrift) [reply incubator/vault/Chart.yaml]: Fixed

**Pull request merged**